Eugene/MovieSearch/my-app/src/components/MainPage/search-panel/SearchPanel.tsx

```diff
@@ -0,0 +1,117 @@
+import { parse, stringify } from "querystring";
+import React from "react";
+import "./SearchPanel.css";
+import { Location } from "history";
+
+interface ISearchPanelProps {
+  handleSearch(input: string, searchBy: string): void;
+  location: Location;
+}
+
+interface ISearchPanelState {
+  input: string;
+  searchBy: SearchBy;
+}
+
+enum SearchBy {
+  Title = "title",
+  Genre = "genres",
+}
+
+class SearchPanel extends React.Component<
+  ISearchPanelProps,
+  ISearchPanelState
+> {
+  state: ISearchPanelState = {
+    input: "",
+    searchBy: SearchBy.Title,
+  };
+
+  componentDidMount() {
+    const searchParams = this.props.location.search.slice(1);
```

Comment:
not sure you really need to slice "?". Please check. I think it will be work without it